apps/amo/decorators.py

```diff
@@ -238,3 +240,25 @@ def wrapper(request, username, *args, **kw):
             username = request.user.username
         return f(request, username, *args, **kw)
     return wrapper
+
+
+def atomic(fn):
+    """Set the transaction isolation level to SERIALIZABLE and then delegate
+    to transaction.atomic to run the specified code atomically. The
+    SERIALIZABLE level will run SELECTs in LOCK IN SHARE MODE when used in
+    conjunction with transaction.atomic.
+    Docs: https://dev.mysql.com/doc/refman/5.6/en/set-transaction.html.
+    """
+    # TODO: Make this the default for all transactions.
+    @functools.wraps(fn)
+    @write
+    def inner(*args, **kwargs):
+        cursor = connection.cursor()
+        try:
+            cursor.execute('SET TRANSACTION ISOLATION LEVEL SERIALIZABLE')
+        except OperationalError:
+            # Looks like we're in a transaction already, too late now.
```

Comment:
can you log this error. I know it's mostly for the tests but it could theoretically happen in prod.